src/interchange/src/avro/envelope_cdc_v2.rs

```diff
@@ -259,11 +189,83 @@ pub fn build_schema(row_schema: serde_json::Value) -> Schema {
 mod tests {
 
     use super::*;
+    use crate::avro::encode_datums_as_avro;
+    use crate::encode::column_names_and_types;
+    use mz_avro::types::Value;
     use mz_avro::AvroDeserializer;
     use mz_avro::GeneralDeserializer;
-    use repr::ScalarType;
+    use repr::{ColumnName, ColumnType, RelationDesc, Row, ScalarType};
+
+    use crate::json::build_row_schema_json;
+
+    /// Collected state to encode update batches and progress statements.
+    #[derive(Debug)]
+    pub struct Encoder {
```

Comment:
This doesn't seem like it needs to be `pub` anymore?

Reply:
Will adjust!